File: jmlparser-jml-tools/src/main/java/com/github/jmlparser/stat/ExpressionCosts.java
package com.github.jmlparser.stat;

import lombok.Data;

/**
 * @author Alexander Weigl
 * @version 1 (02.06.22)
 */
@Data
public class ExpressionCosts {
    private int sum = 1;
    private int minus = 1;
    private int divide = 1;
    private int mult = 1;
    private int mod = 1;
    private int land = 1;
    private int lor = 1;
    private int lnot = 1;
    private int band = 1;
    private int bor = 2;
    private int bnot = 1;
    private int quantor = 3;
    private int binderCostsPerVariable = 1;
    private int nullLiteral = 0;
    private int charLiteral = 0;
    private int stringLiteral = 0;
    private int integerLiteral = 0;
    private int longLiteral = 0;
    private int name = 0;
    private int methodCall = 2;
    private int conditional = 2;
    private int cast = 1;
    private int assign = 1;
    private int let = 1;
    private int compare = 1;
    private int _instanceof = 1;
}

File: tools/stat/src/main/kotlin/io/github/jmltoolkit/stat/ExpressionCosts.kt
package io.github.jmltoolkit.stat


/**
 * @author Alexander Weigl
 * @version 1 (02.06.22)
 */
data class ExpressionCosts(
    val sum: Int = 1,
    val minus: Int = 1,
    val divide: Int = 1,
    val mult: Int = 1,
    val mod: Int = 1,
    val land: Int = 1,
    val lor: Int = 1,
    val lnot: Int = 1,
    val band: Int = 1,
    val bor: Int = 2,
    val bnot: Int = 1,
    val quantor: Int = 3,
    val binderCostsPerVariable: Int = 1,
    val nullLiteral: Int = 0,
    val charLiteral: Int = 0,
    val stringLiteral: Int = 0,
    val integerLiteral: Int = 0,
    val longLiteral: Int = 0,
    val name: Int = 0,
    val methodCall: Int = 2,
    val conditional: Int = 2,
    val cast: Int = 1,
    val assign: Int = 1,
    val let: Int = 1,
    val compare: Int = 1,
    val instanceof: Int = 1
)